public/main.js

```diff
@@ -132,31 +145,42 @@ function deleteCompletedTodos() {
             return todo.isComplete;
         });
         completedTodoList.forEach(function(todo) {
-            var createRequest = new XMLHttpRequest();
-            createRequest.open("DELETE", "/api/todo/" + todo.id);
-            createRequest.setRequestHeader("Content-type", "application/json");
-            createRequest.onload = function() {
-                if (this.status === 200) {
+            fetch("/api/todo/" + todo.id, {
+                method: "DELETE",
+                headers: {
+                    "Content-type": "application/json"
+                }
+            }).then(function(response) {
+                if (response.status === 200) {
                     reloadTodoList();
                 } else {
-                    error.textContent = "Failed to delete. Server returned " + this.status + " - " + this.responseText;
+                    throw Error("Failed to delete. Server returned " + response.status + " - " + response.statusText);
                 }
-            };
-            createRequest.send();
+            }).catch(function(err) {
+                error.textContent = err;
+            });
         });
     });
 }
 
 function completeTodo(todo) {
-    var createRequest = new XMLHttpRequest();
-    createRequest.open("PUT", "/api/todo/" + todo.id);
-    createRequest.setRequestHeader("Content-type", "application/json");
-    createRequest.onload = function() {
-        if (this.status === 200) {
+    fetch("/api/todo/" + todo.id, {
+        method: "PUT",
+        headers: {
+            "Content-type": "application/json"
+        },
+        body: JSON.stringify({
+            isComplete: !todo.isComplete
+        })
+    }).then(function(response) {
+        if (response.status === 200) {
             reloadTodoList();
+        } else {
+            throw Error("Failed to complete. Server returned " + response.status + " - " + response.statusText);
         }
-    };
-    createRequest.send(JSON.stringify({isComplete: !todo.isComplete}));
+    }).catch(function(err) {
```

Comment:
this is the same for every service call. It can be extracted to something like `.catch(errorHandler)`